common/lib/nft/storage/nft-uri.ts

```diff
@@ -12,6 +12,14 @@ const nftStorageSet = (storage: string): void => {
   }
 };
 
+const optionalFieldSet = (optFieldName: string, optField: string) => {
```

Comment:
do not create another function, but change option string parameter in nftStorageSet  to object

something with this type

```typescript
type OptionType = {
  storage : "ipfs" | "swarm" | "arweave"; 
  apiEndpoint: string;
  apiKey: string
  gateway: string;
}
```

with this in config 
```json
for IPFS
 {
  "storage" : "ipfs",
  "apiEndpoint" : "https://api.nft.storage",
  "apiKey":  {{ipfsNftStorageKey}}
  "gateway": "https://nftstorage.link/ipfs",
}
for SWARM
 {
  "storage" : "swarm",
  "apiEndpoint" : {{swarmEndpoint}},
  "apiKey":  {{swarmBatchId}}
  "gateway": "https://api.gateway.ethswarm.org/bzz",
}
```

Reply:
done